.travis.yml

```diff
@@ -23,13 +41,20 @@ install:
   # cabal and ghc tries to use them all. Which is bad idea on a shared box.
   # See also: https://ghc.haskell.org/trac/ghc/ticket/9221
   - sed -i 's/^jobs:/-- jobs:/' ${HOME}/.cabal/config
-  - mkdir -p sandboxes/$GHCVER/${STACKAGE:-none}
-  - cabal sandbox init --sandbox sandboxes/$GHCVER/${STACKAGE:-none}
-  # Download stackage cabal.config
-  # Not sure whether filtering out purescript is necessary
+  # Cache sandboxes in ~/cabal-sandboxes
+  # Move right sandbox to .cabal-sandbox if exists
+  - if [ -d ~/cabal-sandboxes/$GHCVER-${STACKAGE:-none} ]; then
```

Comment:
@paf31 no, the `compiler` trick in build matrix does the job. In https://github.com/travis-ci/travis-ci/issues/4393 is an issue, and this hackish - workaround mentioned.